src/WebJobs.Extensions.SendGrid/Config/SendGridOptions.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.WebJobs.Extensions.Bindings;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Options;
+using SendGrid.Helpers.Mail;
+
+namespace Microsoft.Azure.WebJobs.Extensions.SendGrid
+{
+    /// <summary>
+    /// Defines the configuration options for the SendGrid binding.
+    /// </summary>
+    public class SendGridOptions
+    {
+        /// <summary>
+        /// Gets or sets the SendGrid ApiKey. If not explicitly set, the value will be defaulted
+        /// to the value specified via the 'AzureWebJobsSendGridApiKey' app setting or the
+        /// 'AzureWebJobsSendGridApiKey' environment variable.
+        /// </summary>
+        public string ApiKey { get; set; }
+
+        /// <summary>
+        /// Gets or sets the default "to" address that will be used for messages.
+        /// This value can be overridden by job functions.
+        /// </summary>
+        /// <remarks>
+        /// An example of when it would be useful to provide a default value for 'to' 
+        /// would be for emailing your own admin account to notify you when particular
+        /// jobs are executed. In this case, job functions can specify minimal info in
+        /// their bindings, for example just a Subject and Text body.
+        /// </remarks>
+        public EmailAddress ToAddress { get; set; }
+
+        /// <summary>
+        /// Gets or sets the default "from" address that will be used for messages.
+        /// This value can be overridden by job functions.
+        /// </summary>
+        public EmailAddress FromAddress { get; set; }
+
+        internal class Setup : IConfigureOptions<SendGridOptions>
+        {
+            private readonly IConfiguration _config;
+
+            public Setup(IConfiguration config)
+            {
+                _config = config;
+            }
+
+            public void Configure(SendGridOptions options)
+            {
+                // TODO: Need to access raw to/from strings from config
```

Comment:
We need to handle the mapping of host.json to/from raw string values to EmailAddress instances here. Need to determine the right way to do this.

Reply:
The goal is to do this in a generic way for extensions so they don’t all need to have setup classes. I’ll have something in place tomorrow.

Reply:
I did something in the Cosmos changes that I wanted to get comments on today -- binding directly to a section of `IConfiguration` (and note that you don't need separate setup classes anymore):

    services.AddOptions<CosmosDBOptions>()
        .Configure<IConfiguration>((options, config) =>
        {
            config.GetSection("CosmosDB")?.Bind(options);
        });

Reply:
Updated to follow this pattern. After this PR gets in, if Fabio has a new model we can move to that. These changes work.